crates/flow-pkt/build.rs

```diff
@@ -29,81 +29,180 @@ fn main() {
     let registry_path = std::path::Path::new(&manifest_dir).join("registry");
     let subregistry_path = registry_path.join("subregistry");
 
-    // FlowDirection SubRegistry Path
-    let flow_direction_path = subregistry_path
-        .join("iana_flow_direction.xml")
-        .into_os_string()
-        .into_string()
-        .expect("Couldn't load flowDirection registry file");
-
-    let nokia_path = registry_path
-        .join("nokia.xml")
-        .into_os_string()
-        .into_string()
-        .expect("Couldn't load nokia registry file");
-    let nokia_source = SourceConfig::new(
-        RegistrySource::File(nokia_path),
-        RegistryType::IanaXML,
-        637,
-        "nokia".to_string(),
-        "Nokia".to_string(),
-        None,
-    );
-
-    // Add any external sub-registries for VMWare
-    let external_subregs = vec![
-        ExternalSubRegistrySource::new(
-            RegistrySource::Http(PROTOCOL_NUMBERS_URL.to_string()),
-            SubRegistryType::ValueNameDescRegistry,
-            String::from("protocol-numbers-1"),
-            880,
-        ),
-        ExternalSubRegistrySource::new(
-            RegistrySource::File(flow_direction_path.clone()),
-            SubRegistryType::ValueNameDescRegistry,
-            String::from("ipfix-flow-direction"),
-            954,
-        ),
-    ];
-    let vmware_path = registry_path
-        .join("vmware.xml")
-        .into_os_string()
-        .into_string()
-        .expect("Couldn't load VMWare registry file");
-    let vmware_source = SourceConfig::new(
-        RegistrySource::File(vmware_path),
-        RegistryType::IanaXML,
-        6876,
-        "vmware".to_string(),
-        "VMWare".to_string(),
-        Some(external_subregs),
-    );
-
-    // Add any external sub-registries for IANA
-    let external_subregs = vec![
-        ExternalSubRegistrySource::new(
-            RegistrySource::Http(PROTOCOL_NUMBERS_URL.to_string()),
-            SubRegistryType::ValueNameDescRegistry,
-            String::from("protocol-numbers-1"),
-            4,
-        ),
-        ExternalSubRegistrySource::new(
-            RegistrySource::File(flow_direction_path),
-            SubRegistryType::ValueNameDescRegistry,
-            String::from("ipfix-flow-direction"),
-            61,
-        ),
-    ];
-    let iana_source = SourceConfig::new(
-        RegistrySource::Http(IPFIX_URL.to_string()),
-        RegistryType::IanaXML,
-        0,
-        "iana".to_string(),
-        "IANA".to_string(),
-        Some(external_subregs),
-    );
-    let configs = Config::new(iana_source, vec![nokia_source, vmware_source]);
-    generate(&out_dir, &configs).unwrap();
-
-    println!("cargo:rerun-if-changed=build.rs");
+    if cfg!(feature = "backwards-compatibility-snapshot") {
```

Comment:
I like the idea, just I would invert the condition to make it more user-friendly. Instead of `backwards-compatibility-snapshot` I would specify `iana-upstream-build` feature. And make the code by default always build from the snapshot xml unless otherwise specified. 
My assumption here that the majority of users  care about well-established IANA entries that already exists for a long time, the ones (such ourselves in NetGauze) who cares about the new IANA entries can opt-in to build directly from IANA registry.

What do you think?

Reply:
That sounds awesome! I did the changes.